engine/consensus/compliance/core.go

```diff
@@ -354,6 +355,20 @@ func (c *Core) OnBlockVote(originID flow.Identifier, vote *messages.BlockVote) e
 	return nil
 }
 
+func (c *Core) OnTimeoutObject(originID flow.Identifier, timeout *model.TimeoutObject) error {
```

Comment:
same ⚠️ ❗ 

On the wire, `SignerID` should _not_ be transmitted. Furthermore, It would be great to have different message types for main consensus vs collector consensus; again following the same patter as [`messages.BlockVote`](https://github.com/onflow/flow-go/blob/c6498416374a5d9674d5a4d333793de12f851cd1/model/messages/consensus.go#L14-L20). You could for instance introduce `message.TimeoutObject` and `message.CollectorTimeout`